File: Two-Sum.js
1/**
2 * @param {number[]} nums
3 * @param {number} target
4 * @return {number[]}
5 */
6var twoSum = function(nums, target) {
7    let map = new Map();
8    for(let i =0;i<nums.length;i++){
9        let need = target - nums[i];
10        if(map.has(need))  return [map.get(need), i];
11        map.set(nums[i],i)
12    }
13    /*
14    for(let i =0;i<nums.length;i++){
15        for(let j=i+1;j<nums.length;j++){
16            if(nums[i]+nums[j]==target) return [i,j]
17        }
18    }
19    */
20};

File: Two-Sum.cs
1public class Solution {
2    public int[] TwoSum(int[] nums, int target) {
3        Dictionary<int,int> map = new Dictionary<int,int>();
4        for(int i =0;i<nums.Length;i++){
5            int need = target - nums[i];
6            if(map.ContainsKey(need))  {
7                return [map.GetValueOrDefault(need), i];
8            }
9            map[nums[i]]=i;
10        }
11        return [];
12    }
13}